Разметка всех страниц

DannyKGZ opened the pull request:
---
:mortar_board: [Разметка всех страниц](https://up.htmlacademy.ru/htmlcss/36/user/815423/tasks/5)

:boom: https://htmlacademy-htmlcss.github.io/815423-sedona-36/2/

DannyKGZ committed:
Главная страница+ половина каталога

Сделал главную страницу и половину каталога. Постараюсь часто коммитить

DannyKGZ committed:
добавляет ссылки + хлебные крошки

добавил ссылки на соц сети
И поменял тег на хлебных крошках

DannyKGZ committed:
catalog - добавляет блок каталога гостиниц

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+<html lang="ru">
+  <head>
+    <meta charset="utf-8">
+    <link rel="stylesheet" href="style/style.CSS">
```
Расширение должно быть записано в правильном регистре, иначе после публикации сайта сервер, который работает обычно не на windows, не найдет эти стили, ведь регистр имен в unix-системах важен.

efiand replied:
И путь неправильный. Так как главной странице тоже нужны стили, папку стилей нужно будет переложить выше. И назвать ее без `.css`, так как расширение уже есть у файла.
Так как страница в подпапке, путь к стилям пойдет через родительский каталог, а именно `../style/style.css`.

Это будет касаться и путей ко всем картинкам. Допустимо хранить страницу рядом с index.html, без подпапки `catalog/`, тогда лишних проблем не будет.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+  </head>
+  <body class="wrapper">
+    <header class="page-header">
+        <img src="#" alt="логотип сайта Седона" class="logo">
```
Логотип традиционно делают ссылкой на главную страницу.

efiand replied:
Для удобства поддержки кода рекомендуется ставить атрибут class первым

efiand replied:
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-01

В данном случае единообразие в следовании кодгайду предполагает, что отступ дочернего элемента отличается только на 2 пробела, а здесь - аж на 4. В VS Code проще всего выравнивать автоматически клавишами `alt + shift + f`.

efiand replied:
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-01

В конце описания должна быть точка для улучшения интонации скринридера.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+        <img src="#" alt="логотип сайта Седона" class="logo">
+        <nav class="header-navigation">
+          <ul class="navigation-list">
+            <li class="navigation-item"><a class="navigation-link">Главная</a></li>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

Эта ссылка не ведет сама на себя, это ссылка на главную страницу, index.html

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+        <nav class="header-navigation">
+          <ul class="navigation-list">
+            <li class="navigation-item"><a class="navigation-link">Главная</a></li>
+            <li class="navigation-item"><a class="navigation-link">О Седоне</a></li>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

Это ссылка не на текущую страницу, при этом страницы такой пока нет, поэтому нужно использовать заглушку `#` или даже `#!`

efiand replied:
Ссылки-заглушки рекомендую писать как `<a href="#!">` по двум причинам:

- Отсутствует скачок к верхней части страницы, что важно для всего, что после хедера, но на проекте все должно быть единообразно, поэтому и тут стоит так сделать.
-  При замене временных ссылок на реальные под замес не попадет реальная ссылка к верху страницы, если она есть на проекте.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+          <ul class="navigation-list navigation-user">
+            <li class="navigation-item">
+              <a href="#" class="navigation-link-search">
+                <span class="visually-hidden">Поиск</span>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-01

В конце описания должна быть точка для улучшения интонации скринридера.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+            </li>
+            <li class="navigation-item">
+              <a href="#" class="navigation-link-heart">
+                <span class="visually-hidden"></span>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-01

* Нужно добавить осмысленный текст - например, `Избранное:`
* Отсутствует видимая контентная информация - `12` (число закладок в избранном). Можно добавить в виде отдельного спана.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+              </a></li>
+            <li class="navigation-list">
+              <a href="#" class="navigation-link-button">
+                <span class="visually-hidden">Хочу сюда!</span>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#TEST-06

Здесь текст не скрыт, `<span>` не нужен.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+        </ul>
+      </header>
+
+      <section class="catalog-products">
```
Судя по названию секции, она нужна после фильтров, а для фильтров секцией уже является `.filter-block`

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+  </header>
+
+  <main class="page-main">
+    <section class="filter-block">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-07

Есть проблема с закрытием секций

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+        <ul class="breadcrumbs">
+          <li class="breadcrumbs-item">
+            <a class="breadcrumbs-link" href="index.html">
+              <img src="#" alt="Иконка главной страницы">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#IMG-01

Это декоративное изображение, будет добавляться позже через CSS.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+          </fieldset>
+          <fieldset class="catalog-filter-group">
+            <legend class="catalog-filter-title">Тип жилья:</legend>
+            <ul class="catalog-filter-list"></ul>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-07

Не в том месте закрыт тег.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+            </li>
+          </fieldset>
+          <fieldset class="catalog-filter-group">
+            <ul class="catalog-filter-list"></ul>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-07

Тег закрыт преждевременно.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+          </fieldset>
+          <fieldset class="catalog-filter-group">
+            <ul class="catalog-filter-list"></ul>
+              <li class="catalog-filter-item">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Если использовать тут список (хотя у меня сомнения, что две не самых похожих кнопки это список) - то должно быть два элемента. Две кнопки в одном li - делают бессмысленным список.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+            <ul class="catalog-filter-list"></ul>
+              <li class="catalog-filter-item">
+                <button class="button" type="submit">Применить</button>
+                <input type="reset" value="Сбросить">
```
Чтобы стилизовать элемент без проблем, используйте современный способ - тег `<button>`.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+              <input class="double-range-price" type="range">
+            </li>
+          </fieldset>
+          <fieldset class="catalog-filter-group">
```
Филдсеты семантически нужны для группировки полей ввода. Кнопки таковыми не являются.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+          </fieldset>
+          <fieldset class="catalog-filter-group">
+            <legend class="catalog-filter-title">Стоимость, ₽:</legend>
+            <li class="catalog-filter-item">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-07

Тег `<li>` без `<ul>`.

---

https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Тег списка из одного элемента, в который запиханы все поля, смысла не имеет.

efiand replied:
Третий блок вообще не инпут, ведь он не несет никакой дополнительной информации, это визуальная максималка для двух полей. Можно пока удалить его, позже будет рассказано, как делать такой блок.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+          <fieldset class="catalog-filter-group">
+            <legend class="catalog-filter-title">Стоимость, ₽:</legend>
+            <li class="catalog-filter-item">
+              <input class="initial-price" placeholder="От" value="0" type="number" name="initial-price">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-02
Плейсхолдеры и значение не могут существовать на одном инпуте. `От` и `До` - это лейблы.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+            <legend class="catalog-filter-title">Стоимость, ₽:</legend>
+            <li class="catalog-filter-item">
+              <input class="initial-price" placeholder="От" value="0" type="number" name="initial-price">
+              <input class="final-price" placeholder="До" value="9000" type="number" name="final-price">
```
Что касается названий полей - это не начальная и окончательная цены, а минимальная и максимальная, в пределах которых мы ищем результаты.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+            <ul class="catalog-filter-list">
+              <li class="catalog-filter-item">
+                <label class="control">
+                  <input class="control-input" type="checkbox" name="swimming-pool" checked>
```
Когда мы выбираем из однородного списка, рекомендуется чекбоксы сделать с одинаковым именем, но с разным value (в них то, что сейчас в name). То есть это будет похоже на группу радиобаттонов. только выбор множественный.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+            <ul class="catalog-filter-list"></ul>
+              <li class="catalog-filter-item">
+                <label class="control">
+                  <input class="control-input" type="radio" name="hotel" checked>
```
Чтобы выбор был в пределах одной группы, нужно каждому инпуту дать одинаковое имя и разное value (туда поместить то, что сейчас в `name`)

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+          </fieldset>
+        </form>
+    </section>
+    <div class="catalog">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Пока мы не приступили к стилизации, лучше не задавать div во избежание избыточности

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+        </form>
+    </section>
+    <div class="catalog">
+      <div class="catalog-filter"></div>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#TEST-06

пустой div тут не нужен, а вот текст `Найдено гостиниц: 38` в семантичном теге - нужен

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+    </section>
+    <div class="catalog">
+      <div class="catalog-filter"></div>
+        <a href="#"></a>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Зачем эта ссылка?

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+          <option value="expensive">Сначала дорогое</option>
+        </select>
+        <ul class="btn-view-list">
+          <li class="btn-view-item"><span class="visually-hidden"></span></li>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-02

Спаны следует обернуть в ссылки или кнопки, дать осмысленное описание с точкой на конце

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+    <h2 class="newsletter-title">Подпишитесь на рассылку</h2>
+    <p class="newsletter-text">Только полезная информация и никакого спама, честное бойскаутское!</p>
+    <form class="newsletter-form" action="https://echo.htmlacademy.ru/" method="post">
+        <label for="newsletter-email">Email</label>
```
Отступы нужно будет поправить по всему проекту

efiand replied:
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-02

Тут тоже точка в конце. Нужно исправить по всему проекту, даже там, где я не добавляю однотипных комментариев.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+        <label for="newsletter-email">Email</label>
+        <input class="field"  placeholder="Ваш email" type="email" name="newsletter-email" id="newsletter-email" required>
+        <button class="button newsletter-button" type="submit">
+          <span class="newsletter-subscribe">Подписаться</span>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Спан не нужен, так как текст видимый и без особого оформления (стили можно прописать на самой кнопке)

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+
+
+  <footer class="page-footer">
+    <div class="social-list">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Враппер для единственного элемента без центровки - лишний.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+  <footer class="page-footer">
+    <div class="social-list">
+      <ul class="social-item">
+        <li class="social-items">
```
items - это множественное число. И логично было бы так назвать весь список, а отдельный элемент - item

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+      </ul>
+    </div>
+    <div class="footer-phone-number">
+      <span>+7 (812) 812-12-12</span>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

Телефон это ссылка. И для стилизации она тут самодостаточно, не надо спана и дива.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+    <div class="footer-phone-number">
+      <span>+7 (812) 812-12-12</span>
+    </div>
+    <div class="footer-logo">
```
И тут див не нужен, ссылка умеет в стили

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+    </div>
+    <div class="footer-logo">
+      <a href="https://htmlacademy.ru/">
+        <img src="#" alt="логотип htmlacademy">
```
https://htmlacademy.ru/about

правильное название компании - `HTML Academy`

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+          <ul class="navigation-list">
+            <li class="navigation-item"><a class="navigation-link">Главная</a></li>
+            <li class="navigation-item"><a class="navigation-link">О Седоне</a></li>
+            <li class="navigation-item"><a class="navigation-link" href="/catalog/catalog.html">Гостиницы</a></li>
```
Эта ссылка может не работать, если сайт опубликован не по адресу домена, а в подпапке. Например, опубликованная версия задания:
![изображение](https://user-images.githubusercontent.com/26799424/215536568-7c650b6e-8152-4d30-b6d1-178dea615d4e.png)

Ссылка, начинающаяся с `/`, ведет не в папку проекта, а на весь домен, и страница не находится.
![изображение](https://user-images.githubusercontent.com/26799424/215536020-ae4e8c8b-5357-491d-923b-75bd759f5703.png)

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
 
-  </body>
+  <main class="page-main">
+    <div class="presentation">
```
У тега main можно использовать тег header. В него вместо div можно положить и эти картинки, и скрытый заголовок первого уровня.

efiand replied:
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-02

Отсутствует `h1`. Если на макете его нет, в учебных целях придумаем сами и сделаем визуально скрытым (не забывая про точку на конце для скринридера)

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+      <img src="#" alt="Надпись названия горы">
+    </div>
+
+    <div class="page-main_story">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-01

Единообразное наименование классов предполагает использование дефисов для разделения слов. Исключение - методология. `_` используется в методологии БЭМ для модификаторов, но не так. Поэтому лучше пока БЭМ не трогать, он будет еще на втором интенсиве

efiand replied:
Вместо двух дивов тут очень бы подошла секция.

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
-  </body>
+  <main class="page-main">
+    <div class="presentation">
+      <img src="#" alt="Надпись приветствия">
```
В alt эмблемы стоит прописать то, что на ней написано

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+  <main class="page-main">
+    <div class="presentation">
+      <img src="#" alt="Надпись приветствия">
+      <img src="#" alt="Надпись названия горы">
```
Горы - на мой взглял, это фон, так как он не особо информативный и скрыт эмблемой. Контентная графика не скрывается. Допустимо делать тут `<img>`, но я оставлю информацию, почему тут подходит и фон.

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+    </div>
+
+    <div class="page-main_story">
+      <h2>Седона — небольшой городок в Аризоне, заслуживающий большего!</h2>
```
Этот текст по смыслу не подходит на роль заголовка, ведь по нему непонятно, про что дальнейшая информация. Это лид-текст, слоган и т. п.

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+      <h2>Седона — небольшой городок в Аризоне, заслуживающий большего!</h2>
+      <p>Рассмотрим причины, по которым Седона круче, чем Гранд-Каньон!</p>
+    </div>
+    <div class="page-main_about">
```
Вместо этого дива очень подходит список из пяти элементов.

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+    <div class="page-main_about">
+      <div class="about-item about-box">
+        <h3>Настоящий городок</h3>
+        <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
```
Далее следует контентное изображение

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+        <h3>Настоящий городок</h3>
+        <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+      </div>
+      <div class="about-item about-box"></div>
```
Столько пустых элементов нам не нужно.

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+      </div>
+    </div>
+
+    <div class="page-main_story">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#TEST-06

То, что оформление одинаковое, теги и классы, это правильно. Но только контент там совсем другой )

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+    </div>
+
+    <div class="advantages">
+      <div class="advantages-wrapper">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Два дива здесь перебор при всем многообразии вариантов стилей сетки.

efiand replied:
И вместо одного дива тут подошел бы список из трех элементов

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+      <div class="advantages-wrapper">
+        <div class="advantages-item about-box">
+          <img src="#" alt="Изображен значок жилья">
+          <h4>Жилье</h4>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-02

нарушена иерархия заголовков: после h2 идет h4.

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+          <p>Рекомендуем пожить в настоящем мотеле, всё как в кино!</p>
+        </div>
+        <div class="advantages-item about-box">
+          <img src="#" alt="Изображен значок еды">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#IMG-01

В том-то и дело, что это "значок", декоративное изображение. Нужно использовать CSS.

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+    <div class="page-main_story">
+      <h2>Заинтересовались?</h2>
+      <p>Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+      <button class="button-search-hostel" type="button"><span>ПОИСК ГОСТИНИЦЫ В СЕДОНЕ</span></button>
```
Спан лишний

efiand replied:
Рекомендуется использовать правильный регистр слов. Капслок интонационно читается как отчаяние или гнев. Прописные буквы можно сделать с помощью стилей.

efiand commented on catalog/catalog.html:
```diff
@@ -0,0 +1,197 @@
+
+
+
+  <section class="newsletter">
```
Надо сделать одинаково на обеих страницах: если секция на главной внутри main, то и в каталоге надо так

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+          <label for="newsletter-email">Email</label>
+          <input class="field"  placeholder="Ваш email" type="email" name="newsletter-email" id="newsletter-email" required>
+          <button class="button newsletter-button" type="submit">
+            <span class="newsletter-subscribe">Подписаться</span>
```
Лишний спан

efiand commented on index.html:
```diff
@@ -2,11 +2,150 @@
+      <button class="button-search-hostel" type="button"><span>ПОИСК ГОСТИНИЦЫ В СЕДОНЕ</span></button>
+    </div>
+
+    <section class="newsletter">
```
Про форму, шапку и подвал см. также замечания из каталога

efiand commented on catalog/catalog.html:
```diff
@@ -133,19 +133,74 @@ <h1 class="inner-header-title">Гостиницы Седоны</h1>
     </div>
     <div class="catalog-card">
```
Тут очень бы список подошел

efiand commented on catalog/catalog.html:
```diff
@@ -133,19 +133,74 @@ <h1 class="inner-header-title">Гостиницы Седоны</h1>
+        <p>Гостиница</p>
+        <p>От 4000 Р</p>
+        <a href="#" class="product-card-button button">Подробнее</a>
+        <a href="#" class="product-card-button button product-card-button-active">Избранное</a>
```
Я бы цвета регулировал не через `product-card-button-*`, а через `button-*`. Ведь синяя и коричневая кнопки - это общая логика, не только в каталоге. А `butotn-active` - это скорее зеленая кнопка

efiand commented on catalog/catalog.html:
```diff
@@ -133,19 +133,74 @@ <h1 class="inner-header-title">Гостиницы Седоны</h1>
+          <span class="active"></span>
+          <span></span>
+        </div>
+        <span>Рейтинг: 8,5</span>
```
Для фразы параграф подходит очень хорошо

efiand commented on catalog/catalog.html:
```diff
@@ -133,19 +133,74 @@ <h1 class="inner-header-title">Гостиницы Седоны</h1>
+        <a href="#" class="product-card-button button">Подробнее</a>
+        <a href="#" class="product-card-button button product-card-button-active">Избранное</a>
+        <div class="rating-card-result">
+          <span class="active"></span>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

декорации не нужно делать через разметку. Тут через фон можно будет звезды расположить, а в разметке нужно значение звездного рейтинга отелей, например:

```html
<p class="star-rating star-rating-4">
  <span class="visually-hidden">4 звезды.</span>
</p>
```

efiand reviewed (commented):
Необходимо следить за единообразием кода и правильным закрытием тегов, а также проверять валидность.

efiand reviewed (commented):
Необходимо следить за единообразием кода и правильным закрытием тегов, а также проверять валидность.

DannyKGZ committed:
додедлывает все доработки

Доделал все замечания. Кроме списков, вы порекомендовали список в блоке page-main_about. Я оставил дивами, так как спозиционирую их потом через гриды

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
-    <link rel="stylesheet" href="style/style.CSS">
-    <title>HTML Academy: Седона</title>
+      <meta charset="UTF-8">
+      <link rel="stylesheet" href="../catalog/style.css/style.css">
```
Снова неверные пути.

Папка style не лежит в папке catalog (у нас не настолько большое количество стилей, чтобы их постранично разделять, к тому же требования известны: https://up.htmlacademy.ru/htmlcss/36/criteries#CSS-01 `Собственные стили проекта должны быть подключены одним файлом`).

Но даже если бь и нужно было обращаться к папке на этом уровне - очень странно подниматься наверх, чтобы потом вернуться обратно :-) К тому же папка названа не `style.css`, а `style`:
![изображение](https://user-images.githubusercontent.com/26799424/216591191-c34263e2-c44f-4311-8715-00df97681e12.png)
(как мы видим по цвету каталога в VS Code, это общепринятый подход, в отличие от того, чтобы писать папкам ненужные расширения)

Поэтому правильный путь - `../style/style.css`.

Почитайте, как их строить, чтобы не испытывать трудностей: https://up.htmlacademy.ru/htmlcss/36/module/3/item/9

efiand replied:
Никто не заставляет хранить catalog.html в папке, его можно положить рядом с index.html.
Тогда можно будет писать более простые пути - и одинаковые с теми, которые на странице index.html. На ровном месте создали лишнюю проблему :-)

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
-    <meta charset="utf-8">
-    <link rel="stylesheet" href="style/style.CSS">
-    <title>HTML Academy: Седона</title>
+      <meta charset="UTF-8">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#PROJ-01

* Кодгайд Академии предполагает 2 пробела
![изображение](https://user-images.githubusercontent.com/26799424/216591613-d71f5d57-cfb3-4bf5-bfb9-50f7ff6c3f6e.png)

* Кодгайд (любой) предполагает единообразие, а у вас тут то 2 пробела, то 4:
![изображение](https://user-images.githubusercontent.com/26799424/216591922-148668a5-1ee5-4ec6-b114-5ceea0ec2bce.png)
Вам пока что одинаково тяжело воспринимать любой код, а я прямо чувствую боль и слезы, насколько было бы проше анализировать код, если бы он был консистентный (единообразный). Так и ваши будущие коллеги, а их время иногда стоит дороже вашего, пока вы джуниор. Вот поэтому в командах принято соблюдать стандарты.

Возможно, вся проблема в том, что вы или руками делаете отступы, или ваш редактор не учитывает настройки файла `.editorconfig` при автоформатировании. Как бы там ни было:

* Форматируйте в VS Code клавишами `alt + shift + f`
* Если еще нет, установите расширение https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig

И тогда будет правильно и аккуратно (это результат нажатия мной `alt + shift + f`, следует нажимать периодически для автоформатирования):
![изображение](https://user-images.githubusercontent.com/26799424/216592975-5327143c-2c36-4bb2-afa5-40ba460acd9e.png)

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
-    <header class="page-header">
-        <img src="#" alt="логотип сайта Седона" class="logo">
+      <header class="page-header">
+        <img class="logo" src="../index.html" alt="логотип сайта Седона." >
```
Тег `<img>` нужен для картинок, а не для подключения других HTML-страниц. Здесь лучше обернуть `<img>` в ссылку.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
-                </label>
+            <ul class="navigation-list">
+              <li class="navigation-item">
+                <a href="../index.html" class="navigation-link">Главная</a>
```
Кодгайд академии не настаивает, но рекомендует атрибут class писать первым. И многие другие кодгайды тоже.

Если у вас эта проблема из-за `emmet` напишите в личку, я подскажу, как его полечить, чтобы ставил class в начало.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
-                  Wi-fi
-                </label>
+              <li class="navigation-item">
+                <a href="#!" class="navigation-link">Гостиницы</a>
```
Это либо ссылка на ту же страницу (`<a class="navigation-link" href="catalog.html">`), либо ссылка без href (так называемая логическая ссылка) с указанием скринридеру, что это текущая страница (`<a class="navigation-link" aria-current="page">`).
Заглушка `#!` тут не подходит, так как тогда возникнет впечатление, что это ссылка не на эту же страницу.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+              <li class="navigation-item">
+                <a href="#!" class="navigation-link-heart">
+                  <span class="visually-hidden">Избранное.</span>
+                  <span class="visually-hidden">12.</span>
```
Этот элемент виден на макете, поэтому тут нужен другой класс. Придумайте название, как-то связанное с избранным.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+                        </li>
+                        <li class="catalog-filter-item">
+                          <label class="control">
+                          <input class="control-input" type="checkbox" name="catalog-item-input" value="parking" checked>
```
Тут хорошо бы использовать то же название, что на остальных элементах этой группы, то есть `catalog-items`

![изображение](https://user-images.githubusercontent.com/26799424/216595008-10760ed5-9011-4d35-8c21-d6f5b5f02026.png)
А то сейчас при отправке формы выглядит странно, что один элемент стоит особняком. И бэкенд с этим не сможет нормально работать.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+                    <ul class="catalog-filter-list">
+                        <li class="catalog-filter-item">
+                          <label class="control">
+                          <input class="control-input" type="radio" name="hotel" checked>
```
Есть ощущение, что часть замечаний не была увидена.
![изображение](https://user-images.githubusercontent.com/26799424/216595286-1bbbba32-01fd-4d4f-966f-0a92445e4849.png)

Здесь - для выбора одного варианта из всей группы надо сделать так, чтобы имя было одинаковым, а value - разным. Сделайте, как с чекбоксами

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+                  </fieldset>
+                  <div class="catalog-filter-group">
+                    <button class="button" type="submit">Применить</button>
+                    <button class="btn-reset" type="reset">Сбросить</button>
```
Когда в одном проекте для кнопочных классов используется и `button`, и `btn`, могут возникать ошибки. Лучше выбрать один вариант и придерживаться его.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+            </select>
+            <ul class="btn-view-list">
+              <li class="btn-view-item"><a class="btn-view-cards"><span class="visually-hidden">Карточный режим.</span></a></li>
+              <li class="btn-view-item"><a class="btn-view-card"></a><span class="visually-hidden">Оконный режим.</span></a></li>
```
Если бы вы форматировали автоматически и придерживались единообразия иерархии отступов (а не писали в одну строку аж три уровня вложенности) - вы бы обязательно заметили, что тег ссылки закрыт дважды.

https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-07
А если бы имели привычку валидировать работу хотя бы перед отправкой, то тоже такой код не попал бы в мир.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+            <p>От 4000 Р</p>
+            <a href="#!" class="button button-product-card ">Подробнее</a>
+            <a href="#!" class="button button-product-card  button-add">Избранное</a>
+            <div class="rating-card-result star-rating-4">
```
Советую здесь придерживаться семантики: `<p>` очень подходит.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+            <div class="rating-card-result star-rating-4">
+              <span class="visually-hidden">4 звезды</span>
+            </div>
+            <span>Рейтинг: 8,5</span>
```
И здесь `<p>` тоже подходит. Если выше использованы параграфы для фраз, то зачем вдруг тут - не так?

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+          <h2 class="newsletter-title">Подпишитесь на рассылку</h2>
+          <p class="newsletter-text">Только полезная информация и никакого спама, честное бойскаутское!</p>
+          <form class="newsletter-form" action="https://echo.htmlacademy.ru/" method="post">
+              <label for="newsletter-email">Email.</label>
```
Этот лейбл скрыт, поэтому класс visually-hidden тут понадобится.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+              <option value="expensive">Сначала дорогое</option>
+            </select>
+            <ul class="btn-view-list">
+              <li class="btn-view-item"><a class="btn-view-cards"><span class="visually-hidden">Карточный режим.</span></a></li>
```
Три ссылки имеют одинаковые свойства, отличие только в пути к иконке. Поэтому рекомендую использовать два класса:
```html
<a class="btn-view btn-view-cards">
```
Но каждая ссылка из этого набора может быть "текущей выборанной" в данный момент, поэтому цвет рамки меняется. За это отвечает еще один класс, который поначалу на первой кнопке, но при выборе другого способа сортировки он перейдет (с помощью js на стороне браузера или с помощью бэкенда, в зависимости от реализации) на другую ссылку. Именно поэтому нельзя привязывать черную рамку ни к одному из классов, отвечающему за иконку или за остальные свойства. И будет так:

```html
<a class="btn-view btn-view-cards btn-view-current">
```

И в ссылках социальных сетей подход с дополнительным классом очень подходит, ведь, кроме адреса файла иконки, прочая стилизация этих кнопок одинакова.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+        </section>
+      </main>
+      <footer class="page-footer">
+        <div class="social-list">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Здесь всё легко стилизуется на самом теге `<ul>`, `<div>` тут лишний.
Пока не приступили к сеткам, о дивах лучше не думать.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+                          </label>
+                        </li>
+                    </ul>
+                  </fieldset>
```
Речь шла об удалении только этой конструкции

![изображение](https://user-images.githubusercontent.com/26799424/216600064-26aa103a-4c14-4d93-a662-c1c566ee41a5.png)

При стилизации сеток она нам уже понадобится.

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
-    </div>
-  </footer>
-</body>
+        <a class="footer-phone-number">+7 (812) 812-12-12</a>
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

В примерах к критерию описано, как делать href для телефона

efiand commented on catalog/catalog.html:
```diff
@@ -1,252 +1,211 @@
+        <div class="social-list">
+            <ul class="social-items">
+              <li class="social-item">
+                  <a href="#!" class="social-vk">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-03

В критерии написано: `Если в макете есть требования, касающиеся адресов ссылок, они должны быть выполнены.`
А в макете требования есть:

![изображение](https://user-images.githubusercontent.com/26799424/216601377-f0e05e85-801d-4dc5-b6e3-81bd99c8f799.png)

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
-    <link rel="stylesheet" href="style/style.CSS">
-    <title>HTML Academy: Седона</title>
+      <meta charset="UTF-8">
+      <link rel="stylesheet" href="../815423-sedona-36/style/style.css">
```
Когда мы будем сдавать проект, у заказчика не будет папки с вашим id студента. И это всё не будет работать. Папка style лежит на том же уровне, значит, адрес прост: `style/style.css`.

Не надо писать `../` везде лишь бы было, это не показатель относительности пути сам по себе, а это конструкция для перехода НАВЕРХ. А зачем нам выходить наверх и потом обратно, если мы просто можем обратиться к СОСЕДНЕЙ папке

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
   <body class="wrapper">
-    <header class="page-header">
-        <img src="#" alt="логотип сайта Седона" class="logo">
+      <header class="page-header">
```
См. замечания к шапке и подвалу в каталоге, многие проблемы повторяются

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
-            <li class="navigation-item"><a class="navigation-link">О Седоне</a></li>
-            <li class="navigation-item"><a class="navigation-link" href="/catalog/catalog.html">Гостиницы</a></li>
+            <li class="navigation-item">
+              <a href="../index.html" class="navigation-link">Главная</a>
```
Это ссылка на себя же, не надо идти на уровень вверх.
Либо `href="index.html"`, либо без `href` вообще.

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
+          <h1 class="visually-hidden">Sedona.</h1>
+          <img src="#!" alt="Welcome to the gorgeous.">
+          <img src="#!" alt="Sedona.">
+          <img src="#!" alt="Because the grand canyon sucks.">
```
Не странно ли звучит фраза, искусственно побитая на три предложения? Это одна картинка

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
+          <img src="#!" alt="Sedona.">
+          <img src="#!" alt="Because the grand canyon sucks.">
+        </header>
+        <div class="page-main-story">
```
Было бы, конечно, полезно потренироваться использовать секции: это требование всё чаще встречается в различных компаниях.

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
+              <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+            </div>
+            <div class="about-item about-box">
+              <img src="#!" alt="Фотография города Седоны">
```
https://up.htmlacademy.ru/htmlcss/36/criteries#ALLY-01

В конце альтернативных текстов ставится точка.

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
+              <p class="about-title">Настоящий городок</p>
+              <p>Седона — не аттракцион для туристов, там течёт своя жизнь</p>
+            </div>
+            <div class="about-item about-box">
```
Второй класс обычно используется для того, чтобы описать отличия от поведения, заданного первым классом. Здесь и у карточек с текстом, и у картинок, два одинаковых класса. К моменту реализации сеток подумайте об этом.

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
-          <p>Всегда заказывайте топовый фирменный бургер, вы не разочаруетесь!</p>
+        <div class="advantages">
+            <div class="advantages-wrapper">
+              <div class="advantages-item about-box">
```
Если у вас есть общий класс для карточек с разным контентом, не называйте его контентно. Классы `about-` уместны в секции (или диве, в вашем случае) `about`, но создают лишнюю путаницу здесь. Назовите более абстрактно и "стилизационно" - например, `card`, `content-card`, `item-card` и т. п.

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
+            <div class="advantages-wrapper">
+              <div class="advantages-item about-box">
+                  <div class="advantages-image">
+                    <span class="visually-hidden">Жильё.</span>
```
Зачем нам два раза писать Жилье?
Если мы говорим о том, что это не контентная картинка, это не значит, что надо сорить в разметку лишними элементами

https://up.htmlacademy.ru/htmlcss/36/criteries#HTML-06

Здесь в дальнейшем мы используем псевдоэлементы в CSS, а эти дивы стоит убрать

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
-          <img src="#" alt="Изображен значок сувениров">
-          <h4>Сувениры</h4>
-          <p>Не только китайского, но и настоящего местного производства!</p>
+        <div class="page-main_story">
```
Названия с `_` нарушают единообразие именования классов. Исключением мог бы быть БЭМ, но с точки зрения БЭМ тут неверное написание. Так что лучше исправить везде `_` на `-`

efiand commented on index.html:
```diff
@@ -1,151 +1,144 @@
+        <div class="page-main_story">
+            <p class="lead-text">Заинтересовались?</p>
+            <p>Укажите предполагаемые даты поездки, и мы покажем вам лучшие предложения гостиниц в Седоне</p>
+            <button class="button-search-hostel" type="button">Поиск гостиницы в седоне</button>
```
Название города пишется с большой буквы

efiand reviewed (commented):
В футере каталога много проблем, которых нет на главной. Не забывайте синхронизировать шапку и подвал.

Работу принимаю, так как лимит из двух попыток исчерпан, а часы вам могут понадобиться. Критические замечания стоит исправить со следующим заданием.
Все принятые пулреквесты можно найти [здесь](https://github.com/htmlacademy-htmlcss/815423-sedona-36/pulls?q=is%3Apr+is%3Aclosed)
